src/gpg-error.m4

```diff
@@ -40,84 +41,93 @@ AC_DEFUN([AM_PATH_GPG_ERROR],
   dnl but do not document this old, inconsistently-named option.
   AC_ARG_WITH(gpg-error-prefix,,
               [gpg_error_config_prefix="$withval"])
+  min_gpg_error_version=ifelse([$1], ,0.0,$1)
+  gpg_error_use_pkg_config=no
 
   if test x"${GPG_ERROR_CONFIG}" = x ; then
      if test x"${gpg_error_config_prefix}" != x ; then
         GPG_ERROR_CONFIG="${gpg_error_config_prefix}/bin/gpg-error-config"
      else
-       case "${SYSROOT}" in
-         /*)
-           if test -x "${SYSROOT}/bin/gpg-error-config" ; then
-             GPG_ERROR_CONFIG="${SYSROOT}/bin/gpg-error-config"
-           fi
-           ;;
-         '')
-           ;;
-          *)
-           AC_MSG_WARN([Ignoring \$SYSROOT as it is not an absolute path.])
-           ;;
-       esac
+       PKG_CHECK_MODULES([GPG_ERROR], [gpg-error >= $min_gpg_error_version], [gpg_error_use_pkg_config=yes])
+       if test x"${gpg_error_use_pkg_config}" = xyes; then
+         PKG_CHECK_VAR([GPG_ERROR_MT_CFLAGS], [gpg-error], [mtcflags])
+         PKG_CHECK_VAR([GPG_ERROR_MT_LIBS], [gpg-error], [mtlibs])
```

Comment:
only in case config script or prefix were not provided explicitly, try to use pkg-config, then fallback to automatic detection of the config script